### PR TITLE
Feature/88 remove destination stream writer

### LINKED ISSUE #88: StreamWriter should not require destination as constructor argument
**Description**
Currently, the `StreamWriter` requires a destination as a constructor argument. However, e.g. writing to Kafka does not require a destination, but rather a topic. 

**Tasks**
- Remove the constructor argument `destination` from `StreamWriter`
- Remove the default implementation of `StreamWriter.getDestination` and implement it in the derived classes.
- Update archetype

**Consequences**
- Currently, the destination directory is removed if the ingestion fails and the directory has been empty before the ingestion. This functionality assumes that the `getDestination` method of the `StreamWriter` returns a path on hdfs. However, this cannot be guaranteed. `getDestination` may return any string and it can't be assumed that it signifies a path.
The functionality is not very useful in practice since the destination folder is empty only at the very first ingestion. Therefore, this functionality will be removed.

### PR DESCRIPTION
**Pull request by kevinwallimann**
Closes #88 
Description: #88

### CODE REVIEW

File: api/src/main/scala/za/co/absa/hyperdrive/ingestor/api/writer/StreamWriter.scala [hunk 1]
```diff
@@ -23,9 +23,13 @@ import za.co.absa.hyperdrive.ingestor.api.manager.StreamManager
...
+abstract class StreamWriter {
 
   def write(dataFrame: DataFrame, streamManager: StreamManager): StreamingQuery
 
-  def getDestination: String = destination
+  /**
+   *
+   * @return a string describing the destination, used for logging
+   */
+  def getDestination: String
```
Comment (jozefbakus) [inline api/src/main/scala/za/co/absa/hyperdrive/ingestor/api/writer/StreamWriter.scala]: Hmm I am not sure about this method here. It is not "functional" stuff used only as description. I should rather provide a default implementation and return child class here.

Comment (kevinwallimann) [reply api/src/main/scala/za/co/absa/hyperdrive/ingestor/api/writer/StreamWriter.scala]: It's the same as `StreamReader.getSourceName`. Maybe I could rename it to `getDestinationName` to clarify that it's only a name and has no functional usage.

Comment (jozefbakus) [reply api/src/main/scala/za/co/absa/hyperdrive/ingestor/api/writer/StreamWriter.scala]: I know that ... but I am against "no functional" aspect of this. Each implementation should log its own stuff. 
For example there is code ` logger.info(s"STARTING ingestion from '${streamReader.getSourceName}' into '${streamWriter.getDestination}' (id = $ingestionId)")
`, information that Ingestion was started is fine but source and destination log should be done in component which is responsible for that.

Comment (jozefbakus) [reply api/src/main/scala/za/co/absa/hyperdrive/ingestor/api/writer/StreamWriter.scala]: I leave it up to you. I am a fine event with your impl.

Comment (kevinwallimann) [reply api/src/main/scala/za/co/absa/hyperdrive/ingestor/api/writer/StreamWriter.scala]: I'm going to change it. I also don't like it

**Pull request merged**